### PR TITLE
feature(Rolling): Upgraded the rolling function and unified functions. We are using `pd.rolling` now.

### PR DESCRIPTION
**Pull request by szemyd**
Closes #

### CODE REVIEW

File: src/krisi/evaluate/metric.py [hunk 1]
```diff
@@ -80,78 +87,47 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return print_metric(self, repr=True)
 
-    def evaluate(self, y: Targets, predictions: Predictions) -> None:
-        assert self.func is not None, "`func` has to be set to calculate Metric result."
-        check_valid_pred_target(y, predictions)
-
+    def evaluation_(self, *args) -> "Metric":
```
Comment (almostintuitive) [inline src/krisi/evaluate/metric.py:90]: is this a mutable thing?
I only encountered underscore at the end of a property when it was a result of a calculation. if it's a more "private-like" method, the underscore usually goes to the beginning

Comment (szemyd) [reply src/krisi/evaluate/metric.py:90]: Yes, it mutates the object. It was previously also private (with dashes in front), but I had to access it from `Group`

Comment (szemyd) [reply src/krisi/evaluate/metric.py:90]: Then its perfect! Because it shouldn't be called by the user but the method should still be exposed.

Comment (almostintuitive) [reply src/krisi/evaluate/metric.py:90]: I mean a single leading dash :) but as you wish!

Comment (szemyd) [reply src/krisi/evaluate/metric.py]: Done

**Pull request merged**